src/lib.rs

```diff
@@ -201,6 +201,68 @@ impl RegistersExt for Registers {
     }
 }
 
+/// `MemoryObject` represents  raw data in memory.
+#[derive(Clone, Debug)]
+pub struct MemoryObject {
+    /// Memory start address
+    pub address: u64,
+    /// Memory contents
+    pub data: Vec<u8>,
+}
+
+impl MemoryObject {
+    /// Construct `MemoryObject` from string
+    ///
+    /// # Arguments
+    ///
+    /// * 'memory' - gdb output string with memory contents (0xdeadbeaf: 0x01 0x02)
+    pub fn from_gdb<T: AsRef<str>>(memory: T) -> error::Result<MemoryObject> {
+        let mut mem = MemoryObject {
+            address: 0,
+            data: Vec::new(),
+        };
+        let mut lines = memory.as_ref().lines();
+        if let Some(first) = lines.next() {
+            // Get start address
+            let mut splited = first.split(':');
+            if let Some(address) = splited.next() {
+                mem.address =
+                    u64::from_str_radix(&address.split_whitespace().next().unwrap()[2..], 16)?;
+            } else {
+                return Err(error::Error::MemoryObjectParse(format!(
+                    "No memory address:{}",
+                    first
+                )));
+            }
+            // Get memory
+            if let Some(data) = splited.next() {
```

Comment:
repeating code, can be a function

Reply:
I think it's not necessary small code, only two times.